odl/trafos/wavelet.py

```diff
@@ -285,26 +373,63 @@ def __init__(self, domain, wavelet, nlevels=None, pad_mode='constant',
         ...     domain=space, nlevels=1, wavelet='haar')
         >>> wavelet_trafo.is_biorthogonal
         True
-        >>> decomp = wavelet_trafo([[1, 1, 1, 1],
-        ...                         [0, 0, 0, 0],
-        ...                         [0, 0, 1, 1],
-        ...                         [1, 0, 1, 0]])
-        >>> print(decomp)
-        [ 1. ,  1. ,  0.5, ...,  0. , -0.5, -0.5]
+        >>> data = [[1, 1, 1, 1],
+        ...         [0, 0, 0, 0],
+        ...         [0, 0, 1, 1],
+        ...         [1, 0, 1, 0]]
+        >>> decomp = wavelet_trafo(data)
+        >>> decomp.shape
+        (16,)
+
+        It is also possible to apply the transform only along a subset of the
+        axes. Here, we apply a 1D wavelet transfrom along axis 0 for each
+        index along axis 1:
+
+        >>> wavelet_trafo = odl.trafos.WaveletTransform(
+        ...     domain=space, nlevels=1, wavelet='haar', axes=(0,))
+        >>> decomp = wavelet_trafo(data)
         >>> decomp.shape
         (16,)
+
+        In general, the size of the coefficients may exceed the size of the
+        input data when the wavelet is longer than the Haar wavelet. This
+        due to extra coefficients that must be kept for perfect reconstruction.
+        No extra boundary coefficients are needed when the edge mode is
+        ``"pywt_periodic"`` and the size along each transformed axis is a
+        multiple of ``2**nlevels``.
+
+        >>> space = odl.uniform_discr([0, 0], [1, 1], (16, 16))
+        >>> space.size
+        256
+        >>> wavelet_trafo = odl.trafos.WaveletTransform(
+        ...     domain=space, nlevels=2, wavelet='db2',
+        ...     pad_mode='pywt_periodic')
+        >>> decomp = wavelet_trafo(np.ones(space.shape))
+        >>> decomp.shape
+        (256,)
+        >>> wavelet_trafo = odl.trafos.WaveletTransform(
+        ...     domain=space, nlevels=2, wavelet='db2', pad_mode='symmetric')
+        >>> decomp = wavelet_trafo(np.ones(space.shape))
+        >>> decomp.shape
+        (387,)
+
+        References
+        ----------
+        .. _signal extension modes:
+           https://pywavelets.readthedocs.io/en/latest/ref/signal-extension-\
+    modes.html
```

Comment:
Last comment: this link is likely broken due to the whitespace at the start of the second line. Best is to just not break the line in the link (we abandoned that elsewhere as well), let the line go over the PEP8 limit and ignore the pep8speaks complaint.